keps/sig-node/kubelet-node-endpoint.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Kubelet Node Endpoint
+authors:
+  - "clamoriniere"
+owning-sig: sig-node
+participating-sigs: []
+reviewers: []
+approvers: []
+creation-date: 2019-05-02
+status: implementable
+---
+
+# Kubelet Node Endpoint
+
+## Motivation
+
+The Kubelet already proposes a set of endpoints (`/pods`, and soon `/metrics/resource/v1*`) that offer useful information for "local" node monitoring. In addition to those endpoints, we would like to introduce a new `/node` endpoint for displaying the node resource information. It will allow daemonset-based monitoring solutions to retrieve node information locally (on the same host) like "labels, annotations" instead of using a call to the API server, which will solve some scalability issues in big clusters.
+
+![kubelet-node-endpoint](https://i.ibb.co/khVg8rd/Untitled-Diagram.png)
+
+## Proposal
+
+Kubelet already caches and sync its corresponding `Node` information from the API-server. The proposed solution opens a new endpoint "/node" that returns the `k8s.io/api/pkg/core/v1.Node` JSON representation of the Kubelet's Node. Only the Kubelet's Node resource is return.
+
+For consistency the endpoint path "/node" looks similar to the "/pods" endpoint (no API versioning).
+
+## Use Cases
+
+* **Pod metrics and logs tagging with Node labels or annotation**: Monitoring workload solutions based on the kubelet API ("/pods" and "/metrics") used to decorate Pod metrics with additional context information; Information like the "Availability zone", "Provider", "Server type" etc are very useful for issue investigation.
+The current workaround is to retrieve that information from the API-Server, that generates a consequent number of requests to the API-Server on a large kubernetes cluster.
+* **Container images monitoring on Node**: Knowing which Images and Disk usage that it represents on a specific node can be useful for Node monitoring.
+* **Node conditions**: Node's health information is available in the “status.Conditions” list, like DiskPressure, MemoryPressure, PIDPressure… All those conditions are useful for triggering Alert monitoring and used “locally” for correlation with Pod misbehavior.
+* **Node Info and characteristic**: 
+   - Knowing the container runtime can be used to configure monitoring checks that interact directly with the container runtime.
+   - Knowing the operating system could simplify the configuration of the host monitoring.
```

Comment:
It might be nice to list out your goals in terms of how privileged the daemonset you have in mind should need to be.  Detecting the OS can obviously be done easily if you are privileged.

Reply:
right. One of the usecase I had in mind, was about Ebpf injection, that require enough privileges to detect the OS.

It can still be useful for adding the OS information as pod metrics label.